From 47027ce287d852fcd1ea8b2f64728829be7e96c6 Mon Sep 17 00:00:00 2001
From: alpha lin <alpha.lin@rock-chips.com>
Date: Wed, 30 Mar 2016 08:40:38 +0800
Subject: [PATCH] iep/rockchip: revise build fail when CONFIG_RK_IOMMU disabled

When CONFIG_RK_IOMMU disabled, iep build will throw out error
information for some mismatch definitions.

Change-Id: I0fb22550eaaebd62523d794e45de7b94fae8db63
Signed-off-by: alpha lin <alpha.lin@rock-chips.com>
---
 drivers/video/rockchip/iep/iep_drv.c | 2 +-
 drivers/video/rockchip/iep/iep_drv.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/iep/iep_drv.c b/drivers/video/rockchip/iep/iep_drv.c
index 7ed253eb9a54..d1a4036870f3 100644
--- a/drivers/video/rockchip/iep/iep_drv.c
+++ b/drivers/video/rockchip/iep/iep_drv.c
@@ -1005,10 +1005,10 @@ static int iep_drv_probe(struct platform_device *pdev)
 	int ret = 0;
 	struct resource *res = NULL;
 	u32 version;
+	struct device_node *np = pdev->dev.of_node;
 #if defined(CONFIG_IEP_IOMMU)
 	u32 iommu_en = 0;
 	struct device *mmu_dev = NULL;
-	struct device_node *np = pdev->dev.of_node;
 	of_property_read_u32(np, "iommu_enabled", &iommu_en);
 #endif
 
diff --git a/drivers/video/rockchip/iep/iep_drv.h b/drivers/video/rockchip/iep/iep_drv.h
index 496e0d04cf15..f0787342fd74 100644
--- a/drivers/video/rockchip/iep/iep_drv.h
+++ b/drivers/video/rockchip/iep/iep_drv.h
@@ -144,8 +144,8 @@ typedef struct iep_service_info {
 
 #ifdef CONFIG_IEP_IOMMU
 	struct ion_client *ion_client;
-	struct device *iommu_dev;
 #endif
+	struct device *iommu_dev;
 } iep_service_info;
 
 struct iep_reg {
-- 
2.35.3

